### DISCUSSION TITLE
Accept std::basic_string with custom allocators in the API instead of enforcing std::string

### DISCUSSION THREAD

**Discussion by mp-t** [ImageMagick/ImageMagick#6730] [Contribute]
The ImageMagick interface accepts a `std::string` in several functions. For example the `Magick::Image` constructor.

It would be useful if these functions would be templated to allow a `std::basic_string<char, T>`.

This would allow strings with custom allocators to work seamlessly with ImageMagick.

**Comment by dlemstra** [comment]
I don't know if we can do that without breaking the existing API/ABI. Do you have some more info about this?

[a comment by an automated account is not shown]
